src/metadata/EntityMetadata.ts

```diff
@@ -746,6 +746,16 @@ export class EntityMetadata {
         if (firstId === undefined || firstId === null || secondId === undefined || secondId === null)
             return false;
 
+        // Optimized version for the common case
+        if (
+          typeof firstId.id === "string" &&
+          typeof secondId.id === "string" &&
+          Object.keys(firstId).length === 1 &&
+          Object.keys(secondId).length === 1
+        ) {
+            return firstId.id === secondId.id;
```

Comment:
Wouldn't it make sense to have this as the first check in `OrmUtils.deepCompare` to benefit from it in all places it's used, such as:

- https://github.com/typeorm/typeorm/blob/a4dec02cc59d3219a29c7be0322af2253e1452dc/src/persistence/subject-builder/OneToOneInverseSideSubjectBuilder.ts#L142
- https://github.com/typeorm/typeorm/blob/a4dec02cc59d3219a29c7be0322af2253e1452dc/src/persistence/subject-builder/OneToManySubjectBuilder.ts#L120

Reply:
Makes sense!

Should I create a helper function `OrmUtils.compareIds()` for this use case?

Reply:
I pushed a commit with `OrmUtils.compareIds()`